libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2657,6 +2697,15 @@ string YulUtilFunctions::storageSetToZeroFunction(Type const& _type)
 			("functionName", functionName)
 			("clearArray", clearStorageArrayFunction(dynamic_cast<ArrayType const&>(_type)))
 			.render();
+		else if (_type.category() == Type::Category::Struct)
+			return Whiskers(R"(
+				function <functionName>(slot, offset) {
+					<clearStruct>(slot)
```

Comment:
Should we check that ``offset`` here is zero and revert otherwise?
I'd hope that this function would always only be called with ``offset`` as zero, the optimizer would inline it and then remove that check again...

Reply:
Sounds good to do it that way, will update now

Reply:
By the way, should it be `revert` or `invalid`?

Reply:
Good question... for now I'd say ``invalid`` and we can keep it in mind for https://github.com/ethereum/solidity/issues/9824